### PR TITLE
not adding OTHER metrics when all values are zero

### PR DESCRIPTION
**Pull request by puneetjaiswal**
not adding OTHER metrics when all values are zero
screen shot 2016-07-18 at 6 36 28 pm

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java [hunk 1]
```diff
@@ -316,7 +316,19 @@ private void parseGroupByTimeDimensionResponse() {
           otherBuilder.addMetric(metricFunctions.get(i).getMetricName(), otherBaseline[i],
               otherCurrent[i]);
         }
-        rows.add(otherBuilder.build());
+
+        if (otherBuilder.row.getMetrics().size() > 0) {
+          // Add only a non zero metric
+          boolean addMetric = false;
+          for (Row.Metric metric : otherBuilder.row.getMetrics()) {
+            if (metric.getCurrentValue() > 0.0 || metric.getBaselineValue() > 0.0) {
```
Comment (raviarin) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: Is metric value gauranteed to be positive? What if we have metrics that have negative values?

Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: these are just values, not the change (current vs baseline). So yes, these should be non negative.

**Comment by kishoreg** [comment]
instead of just checking for >0, can we check for certain percentage of total value. Even having .1% threshold will remove lot of noise. I think we had this check may be it got lost during re-factoring.

**Comment by puneetjaiswal** [comment]
sure.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java [hunk 2]
```diff
@@ -321,7 +322,10 @@ private void parseGroupByTimeDimensionResponse() {
           // Add only a non zero metric
           boolean addMetric = false;
           for (Row.Metric metric : otherBuilder.row.getMetrics()) {
-            if (metric.getCurrentValue() > 0.0 || metric.getBaselineValue() > 0.0) {
+            // If absolute percentage change is more than default threshold for OTHER
+            if (Utils
```
Comment (raviarin) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: It might be a good time to start adding unit tests. Can you try adding one?

Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: I have been thinking about it. Tests require setting up some boilerplate code (Datagenerator etc). I would do it as separate commit.

**Comment by raviarin** [comment]
LGTM
